packages/transaction-controller/src/helpers/EtherscanRemoteTransactionSource.test.ts

```diff
@@ -126,9 +136,13 @@ describe('EtherscanRemoteTransactionSource', () => {
 
       const remoteSource = new EtherscanRemoteTransactionSource();
 
-      // TODO: Replace `any` with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      await remoteSource.fetchTransactions({} as any);
+      const promiseForFetchTransactions = remoteSource.fetchTransactions(
+        // TODO: Replace `any` with type
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        {} as any,
+      );
+      await advanceTime({ clock, duration: 5000 });
```

Comment:
My understanding was we wanted to remove all `sinon` usages and instead rely solely on `jest`.

Could we achieve the same with `jest.useFakeTimers` and `jest.advanceTimersByTime`?

Reply:
We do want to switch from Sinon to Jest, but this is a case in which we cannot use Jest unfortunately. The `advanceTime` helper makes use of `tickAsync` from Sinon, because we need to flush promises as we are advancing time. The equivalent in Jest would be `jest.advanceTimersByTimeAsync`, but [this method was added in Jest 28](https://jestjs.io/blog/2022/04/25/jest-28#fake-timers), and we are still using Jest 27. We have a ticket tracking the upgrade to Jest 29 [here]( https://github.com/MetaMask/core/issues/3670), so once that's done we can go through and change this helper to use Jest. But in the meantime we still have to use Sinon.